pkg/transport/generic_transport_controller.go

```diff
@@ -0,0 +1,512 @@
+	return c.updatePlacementDecision(ctx, placementDecision, func(placementDecision *v1alpha1.PlacementDecision) bool {
+		return addFinalizer(placementDecision, transportFinalizer)
+	})
+}
+
+func (c *genericTransportController) updateWrappedObjectsAndFinalizer(ctx context.Context, placementDecision *v1alpha1.PlacementDecision) error {
+	if err := c.addFinalizerToPlacementDecision(ctx, placementDecision); err != nil {
+		return fmt.Errorf("failed to add finalizer to PlacementDecision object '%s' - %w", placementDecision.GetName(), err)
+	}
+
+	objectsToPropagate, err := c.getObjectsFromWDS(ctx, placementDecision)
+	if err != nil {
+		return fmt.Errorf("failed to get objects to propagate to WECs from PlacementDecision object '%s' - %w", placementDecision.GetName(), err)
+	}
+
+	wrappedObject, err := convertObjectToUnstructured(c.transport.WrapObjects(objectsToPropagate))
+	if err != nil {
+		return fmt.Errorf("failed to wrap objects to a single wrapped object - %w", err)
+	}
+	// wrapped object name is (PlacementDecision.GetName()-WdsName).
+	// pay attention - we cannot use the PlacementDecision object name, cause we might have duplicate names coming from different WDS spaces.
+	// we add WdsName to the object name to assure name uniqueness,
+	// in order to easily get the origin PlacementDecision object name and wds, we add it as an annotations.
+	wrappedObject.SetName(fmt.Sprintf("%s-%s", placementDecision.GetName(), c.wdsName))
+	setAnnotation(wrappedObject, originOwnerReferenceAnnotation, placementDecision.GetName())
+	setAnnotation(wrappedObject, originWdsAnnotation, c.wdsName)
+
+	for _, destination := range placementDecision.Spec.Destinations { // TODO need to revisit the destination struct and see how to use it properly
+		if err := c.createOrUpdateWrappedObject(ctx, destination.ClusterId, wrappedObject); err != nil {
+			return fmt.Errorf("failed to propagate wrapped object '%s' to all required WECs - %w", wrappedObject.GetName(), err)
+		}
+	}
+
+	return nil
+}
+
+func (c *genericTransportController) getObjectsFromWDS(ctx context.Context, placementDecision *v1alpha1.PlacementDecision) ([]*unstructured.Unstructured, error) {
+	objectsToPropagate := make([]*unstructured.Unstructured, 0)
+	// add cluster-scoped objects to the 'objectsToPropagate' slice
+	for _, clusterScopedObject := range placementDecision.Spec.Workload.ClusterScope {
+		if clusterScopedObject.ObjectNames == nil {
+			continue // no objects from this gvr, skip
+		}
+		gvr := schema.GroupVersionResource{Group: clusterScopedObject.Group, Version: clusterScopedObject.Version, Resource: clusterScopedObject.Resource}
+		for _, objectName := range clusterScopedObject.ObjectNames {
+			object, err := c.wdsDynamicClient.Resource(gvr).Get(ctx, objectName, metav1.GetOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("failed to get required cluster-scoped object '%s' with gvr %s from WDS - %w", objectName, gvr, err)
```

Comment:
This reaction is a little harsh. This `Get` is happening some time after something else determined that the object should be propagated, and the user might have deleted the object in the meantime.